src/CartBlock.C

```diff
@@ -98,22 +88,22 @@ void CartBlock::getInterpolatedData(int *nints,int *nreals,int **intData,
 
         for(i=0;i<listptr->nweights;i++)
         {
-          get_amr_index_xyz(listptr->inode[3*i],listptr->inode[3*i+1],listptr->inode[3*i+2],
-            dims[0],dims[1],dims[2],nf,
-            xlo,dx,index,xtmp);
+          //Q[nq,nZ+2*nf,nY+2*nf,nX+2*nf]--> C++ Cell storage
+          index = (dims[1]+2*nf)*(dims[0]+2*nf)*(listptr->inode[3*i+2]+nf)
+              + (dims[0]+2*nf)*(listptr->inode[3*i+1]+nf)
+              + (listptr->inode[3*i]+nf);
+
           for(n=0;n<nvar;n++)
           {
             weight=listptr->weights[i];
-            qq[n]+=qcell[index[0]+d3nf*n]*weight;
+            qq[n]+=qcell[index+d3nf*n]*weight;
```

Comment:
This line assumes that the fields are stored as columns (if following fortran ordering, is that the case for qcell ?), i.e. all of first field comes first and second field after and so on..
qcell = qcell(ndof,nvar) !f90
        = qcell[nvar][ndof] !c

Reply:
@jsitaraman I suppose this should be controllable at the API level in TIOGA, but for AMReX the solution array is fortran ordered. So 

```
qcell(IMAX, JMAX, KMAX, NQ) ! Fortran
```

Reply:
@jsitaraman Currently all `cart` solution is assumed to be arranged column-wise and is hence consistent with the rest of the code. This can be modified in a future commit throughout the code and should ideally be done with a test-case on hand, but as @sayerhs pointed out, AMReX solution is ordered column-wise.

Reply:
So IMAX = dim[0]+2*nf, yes ?

Reply:
Also is IBLANK => IBLANK(IMAX,JMAX,KMAX)

Reply:
Yes, that's correct.